sdk/identity/azure-identity/src/main/java/com/azure/identity/OnBehalfOfCredentialBuilder.java

```diff
@@ -146,19 +161,16 @@ public OnBehalfOfCredentialBuilder userAssertion(String userAssertion) {
     public OnBehalfOfCredential build() {
         ValidationUtil.validate(CLASS_NAME, LOGGER, "clientId", clientId, "tenantId", tenantId);
 
-        if (clientSecret == null && clientCertificatePath == null) {
-            throw LOGGER.logExceptionAsWarning(new IllegalArgumentException("At least client secret or certificate "
-                + "path should provided in OnBehalfOfCredentialBuilder. Only one of them should "
-                + "be provided."));
-        }
-
-        if (clientCertificatePath != null && clientSecret != null) {
-            throw LOGGER.logExceptionAsWarning(new IllegalArgumentException("Both client secret and certificate "
-                + "path are provided in OnBehalfCredentialBuilder. Only one of them should "
-                + "be provided."));
+        if  ((clientSecret == null && clientCertificatePath == null && clientAssertionSupplier == null)
+            ||(clientSecret != null && clientCertificatePath != null)
```

Comment:
suspect the linter will catch this but if not.
```suggestion
            || (clientSecret != null && clientCertificatePath != null)
```